app/vmselect/multitenant/filters.go

```diff
@@ -0,0 +1,68 @@
+package multitenant
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/auth"
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/prompbmarshal"
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/promrelabel"
+)
+
+// ApplyFiltersToTenants applies the given filters to the given tenants.
+// It returns the filtered tenants.
+func ApplyFiltersToTenants(tenants, filters []string) ([]*auth.Token, error) {
```

Comment:
I think, behavior must be consistent with vminsert - if neither vm_account_id nor  vm_project_id is defined for filters - execute request to the 0:0 tenant (fast path)